### PR TITLE
Remove check for existing term when parsing category via CSV & instead catch an existing term error when adding new category to the database

### LINKED ISSUE #26665: Symbols like ; or . in product category names break the import
**Describe the bug**
Symbols like ; or . in product category names can break the import

**To Reproduce**
1. Export a .csv file
2. In wp-admin, delete the product, remove from trash.
3. Change the category name (and add a symbol like . or ; at the end).
4. If there is no category with the same name but no semicolon or a dot - the category is created.
5. If there is already a product category with the same name except for a dot or a semicolon - the product will be added to that category.

**Expected behavior**
A new product category should be created every time when there's an additional dot or semicolon

**Reported in**

3017949-zen

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by juliaamosova**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/26665. 

Currently, when products are being imported via CSV file, one of the following happens:

- If there is no category with the same name but no semicolon or a dot exists (for example, `Clothing;`) - the category is created when CSV import is completed;
- If there is already a product category with the same name except for a dot or a semicolon exists (for example `Clothing` exists and we are trying to import `Clothing;`) - the product will be added to the existing category `Clothing`. `Clothing;` won't get created. 

<hr>

The reason why the above happens is that we rely on WordPress's functions `term_exists()` and `wp_insert_term()` to check for existing categories during import and add a new term (category) to the database if the category doesn't exist. Both of these functions sanitize the initial entry differently. 

See https://github.com/woocommerce/woocommerce/issues/26665#issuecomment-639145714 for my explanation of what happens in each case in code. 

<hr>

Since `term_exists()` and `wp_insert_term()` are separated from each other and run different queries, for consistency we should not use `term_exists()`. Instead, we should call `wp_insert_term()` and catch error if term exists. Which is what this PR does. 

### How to test the changes in this Pull Request:

1. Before testing this PR, create 2 products with category `Music`.
2. Export products via CSV using a built-in product Importer/Exporter.
3. Delete one of the products completely from the site (including Trash).
3. Open CSV file and edit a category of the deleted product to `Music;`.
4. Add more row for the new product. Fill out all the needed details. Assign it category `Dance;`. 
4. Import edited CSV file back to the site. 
5. Note that category `Music;` didn't get created. Instead, the product that had category `Music;` assigned to it in CSV got added to the existing category `Music`.
6. Note that category `Dance;` did get created. 
7. Now check out this branch and repeat steps 1-6. 
8. Upon importing products via CSV back to the site, both categories `Music;` and `Dance;` should be created in addition to the existing category `Music`. 

<hr>

Pinging @peterfabian and @claudiosanches here for review since I spoke to both of you regarding this issue but anyone else can review as well of course. 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Fix - Remove check for the existing term when parsing category via CSV & instead catch an existing term error when adding a new category to the database

### CODE REVIEW

File: includes/import/class-wc-product-csv-importer.php [hunk 1]
```diff
@@ -384,7 +384,7 @@ public function parse_tax_status_field( $value ) {
 	 *
 	 * @param string $value Field value.
 	 *
-	 * @return array of arrays with "parent" and "name" keys.
+	 * @return array|WP_Error
```
Comment (rrennick) [inline includes/import/class-wc-product-csv-importer.php:387]: @juliaamosova The function doesn't return a `WP_Error` on any flow.

**Review by rrennick — CHANGES_REQUESTED** [review]
@juliaamosova This worked great and should be good to go with the PHPDoc change.

File: includes/import/class-wc-product-csv-importer.php (hunk 1 above)
Comment (juliaamosova) [reply includes/import/class-wc-product-csv-importer.php]: Thanks @rrennick! This was added by PhpStorm automatically when I was editing the code. My bad for not catching this during commit. Fixed.

**Comment by juliaamosova** [comment]
>@juliaamosova This worked great and should be good to go with the PHPDoc change.

Thanks @rrennick! It is ready for re-review.

**Review by rrennick — APPROVED** [review]
@juliaamosova Looks great 🚢

**Pull request merged**